pya/aserver.py

```diff
@@ -287,5 +292,18 @@ def _play_callback(self, in_data, frame_count, time_info, flag):
     def stop(self):
         self._stop = True
 
-    def __del__(self):
+    def _terminate_backend(self):
         self.backend.terminate()
+
+    def __enter__(self):
+        return self.boot()
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        if self._is_active:
+            self.quit()
+        self._terminate_backend()
+
+    def __del__(self):
+        if self._is_active:
+            self.quit()
+        self._terminate_backend()
```

Comment:
as mentioned earlier this could be `self.backend.terminate()` without an extra wrapping unless `_terminare_backend` is planned to do more in the foreseeable future.